[Snyk] Security upgrade werkzeug from 2.2.3 to 3.1.4

oabdoun opened the pull request:
![snyk-top-banner](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### Snyk has created this PR to fix 1 vulnerabilities in the pip dependencies of this project.

#### Snyk changed the following file(s):

- `requirements.txt`



<details>
<summary>⚠️ <b>Warning</b></summary>

```
tensorboard 2.11.2 requires werkzeug, which is not installed.
tensorboard 2.11.2 has requirement setuptools>=41.0.0, but you have setuptools 40.5.0.
scikit-learn 1.0.2 requires scipy, which is not installed.
Resemblyzer 0.1.3 requires scipy, which is not installed.
rdflib 6.3.2 has requirement importlib-metadata<5.0.0,>=4.0.0; python_version >= "3.7" and python_version < "3.8", but you have importlib-metadata 6.7.0.
rdflib 6.3.2 has requirement isodate<0.7.0,>=0.6.0, but you have isodate 0.7.2.
librosa 0.10.0 requires scipy, which is not installed.
```

</details>






---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.
> - Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiI5ZjRmYjBjNS0yMGY1LTQyNGItYmQ5YS1hNWM4YTBhMzYzNTEiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjlmNGZiMGM1LTIwZjUtNDI0Yi1iZDlhLWE1YzhhMGEzNjM1MSJ9fQ==" width="0" height="0"/>
🧐 [View latest project report](https://app.snyk.io/org/video-and-audio/project/59e066d9-266e-46ba-a5d4-a730b414ec1f?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/video-and-audio/project/59e066d9-266e-46ba-a5d4-a730b414ec1f?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Learn about vulnerability in an interactive lesson of Snyk Learn.](https://learn.snyk.io/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"breakingChangeRiskLevel":null,"FF_showPullRequestBreakingChanges":false,"FF_showPullRequestBreakingChangesWebSearch":false,"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"werkzeug","from":"2.2.3","to":"3.1.4"}],"env":"prod","issuesToFix":["SNYK-PYTHON-WERKZEUG-14151620"],"prId":"9f4fb0c5-20f5-424b-bd9a-a5c8a0a36351","prPublicId":"9f4fb0c5-20f5-424b-bd9a-a5c8a0a36351","packageManager":"pip","priorityScoreList":[44],"projectPublicId":"59e066d9-266e-46ba-a5d4-a730b414ec1f","projectUrl":"https://app.snyk.io/org/video-and-audio/project/59e066d9-266e-46ba-a5d4-a730b414ec1f?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","pr-warning-shown","priorityScore"],"type":"auto","upgrade":[],"vulns":["SNYK-PYTHON-WERKZEUG-14151620"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

snyk-bot committed:
fix: requirements.txt to reduce vulnerabilities


The following vulnerabilities are fixed by pinning transitive dependencies:
- https://snyk.io/vuln/SNYK-PYTHON-WERKZEUG-14151620

rmcpantoja commented:
Hi.

Torchaudio also needs to be fixed.

Version 2.9 had many code breaking changes, too many interface changes for a whole refactor. @alexteua any interest on this?

cheers.

snyk-bot committed:
fix: requirements.txt to reduce vulnerabilities


The following vulnerabilities are fixed by pinning transitive dependencies:
- https://snyk.io/vuln/SNYK-PYTHON-WERKZEUG-14151620
